### ISSUE TITLE
compatibility with ev_smart_charging plugin

### ISSUE THREAD

**Issue by acejoh** [andreas-berg/hass-entsoe-dayahead#4]
Hi, 

This version of the integration works very well, thank you for creating it!
I'd like to request restoring compatibility with the [ev_smart_charging](https://github.com/jonasbkarlsson/ev_smart_charging) plugin. 
It looks for the specific structure of the electricity prices, which apparently no longer matches. If I add the sensor with today's (or tomorrow's prices) I get the following error:
<img width="355" alt="image" src="https://github.com/andreas-berg/hass-entsoe-dayahead/assets/18048545/aa4cfbc1-f03f-4828-8b78-c4d2e9a91114">

Thank you for looking into the issue!

**Comment by andreas-berg** [comment]
Hi acejoh,
Sounds like a good idea.. I'll have a look at it.. 
Probably this integration will require some change on the ev_smart_charging-side but I'll try to set it up locally..

**Comment by andreas-berg** [comment]
Yeah.. ev_smart_charging sounds really nice but my old EV charger is not yet smart enough to provide a SoC-sensor, so I'm not currently able to install/test the ev_smart_charging-integration.

I don't think it's a good idea for this hass-entsoe-dayahead-component to implement support the old unmaintained interface from JaccoR, but it should be quite easy for Jonas to do the minor changes the client code. 

@acejoh, could you perhaps open a feature request on https://github.com/jonasbkarlsson/ev_smart_charging, requesting this integration to work.

I'll help Jonas if needed..

**Comment by acejoh** [comment]
I'll add the request but I'm afraid it might take quite some time, he'll probably want to create a generic, user definable template instead. That's why I thought I'd try requesting this from your side :)
But thanks for looking into it nonetheless!

**Comment by andreas-berg** [comment]
Aah, although I had a new HomeAssistant revelation yesterday, my bimmer is available through the BMW Connected component so I now have the SoC-sensor and I could install the ev_smart_charging. 
I see your problem, still not going to change the interface of this component, but it should be possible to configure a template-sensor (in yaml) for the prices-data and modify it to the format that ev_smart_charging expects. 
I'll test this today and get back to you.

**Comment by acejoh** [comment]
Hi andreas, good to hear that! My hybrid BMW should arrive in a few weeks so hopefully I'll be ready by then ^^
Before you spend time on a template sensor, I believe I read that the ev_smart_charging plugin does not support template sensors: https://github.com/jonasbkarlsson/ev_smart_charging/issues/231#issuecomment-1872503814
